src/components/card/card-footer/card-footer.test.tsx

```diff
@@ -0,0 +1,68 @@
+import React from "react";
+import { render, screen } from "@testing-library/react";
+import CardFooter, { CardFooterProps } from ".";
+import CardContext from "../__internal__/card.context";
+
+test("when variant prop is `transparent`, render with transparent background", () => {
+  render(
+    <CardFooter variant="transparent" data-role="card-footer">
+      <div id="non-interactive">View Stripe Dashboard</div>
+    </CardFooter>
+  );
+
+  const cardFooterElement = screen.getByTestId("card-footer");
+  expect(cardFooterElement).toHaveStyle({
+    "background-color": "transparent",
+  });
+});
+
+test.each<CardFooterProps["roundness"]>(["default", "large"])(
+  "renders with the expected border radius styling when roundness is %s",
+  (roundness) => {
+    render(
+      <CardContext.Provider
+        value={{
+          roundness,
+          spacing: "medium",
+        }}
+      >
+        <CardFooter data-role="card-footer">foo</CardFooter>
+      </CardContext.Provider>
+    );
+
+    const cardFooterElement = screen.getByTestId("card-footer");
+    expect(cardFooterElement).toHaveStyle({
```

Comment:
comment: I think toHaveStyle returns true no matter what when token values are asserted so we might need to use toHaveStyleRule in the interim